models/tasks.js

```diff
@@ -575,6 +579,51 @@ const getOverdueTasks = async (days = 0) => {
   }
 };
 
+const updateTaskStatus = async () => {
+  try {
+    const snapshot = await tasksModel.where("status", "==", "COMPLETED").get();
+    const tasksStatusCompleted = [];
+    let summary = {
+      totalTasks: snapshot.size,
+      totalUpdatedStatus: 0,
+      totalOperationsFailed: 0,
+      updatedTaskDetails: [],
+      failedTaskDetails: [],
+    };
+
+    if (snapshot.size === 0) {
+      return summary;
+    }
+
+    snapshot.forEach((task) => {
+      const id = task.id;
+      const taskData = task.data();
+      tasksStatusCompleted.push({ ...taskData, id });
+    });
+    const taskStatusCompletedChunks = chunks(tasksStatusCompleted, DOCUMENT_WRITE_SIZE);
+
+    for (const tasks of taskStatusCompletedChunks) {
+      const res = await updateTaskStatusToDone(tasks);
```

Comment:
The DOCUMENT_WRITE_SIZE is set to 500. Ultimately, it will loop over each document, update it, and only then proceed to the next document. Wouldn't it be better if we reduced it to 100 and updated 100 * 5 concurrently?

<pre>

const res = []
for (const tasks of taskStatusCompletedChunks) {
      res.push(updateTaskStatusToDone(tasks))
}
const batchResult = await Promise.all(res)
//You can update summary through each index

</pre>

The current implementation may cause server timeout.

Reply:
A DOCUMENT_WRITE_SIZE of 500 ensures a single network call to Firestore, offering better efficiency.

However, I understand your concern about potential server timeouts. To address this, you've proposed dividing the batch size to 100 and updating 100 * 5 documents concurrently. While this approach increases the number of network calls, Which is inefficient. As the number of network calls increases, the overall time to complete the operation grows and number of error also increase (In my case I think I have to update more than 3000 document )

It's crucial to note that in the batch operation, if any individual operation fails, it won't trigger a rollback, leaving the already updated documents unchanged. 

But your concern is also non avoidable and server timeout totally depend on firestore execution timeout